### PR TITLE
Task/assignment5

### PR DESCRIPTION
**Pull request by shaksham08**
- Added search functionality to the list

image
image

### CODE REVIEW

File: Week3/Assignment5/src/App.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import SearchBar from "./Components/SearchBar";
+
+const App = () => {
+  const [allUserData, setAllUserData] = useState(teammates);
+
+  return (
+    <>
+      <Header />
+      <SearchBar setAllUserData={setAllUserData} />
+      {(!allUserData || !allUserData.length) && (
```
Comment (nitishnivedan) [inline Week3/Assignment5/src/App.js:16]: can use a ternary here for handling with with 0 length and more than 0

File: Week3/Assignment5/src/Components/SearchBar/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import "./style.css";
+
+const SearchBar = ({ setAllUserData }) => {
+  const [searchTerm, setSearchTerm] = useState("");
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const filteredList =
+      teammates &&
+      teammates.filter((teamData) =>
+        teamData.name.toLowerCase().includes(searchTerm.toLowerCase())
```
Comment (ap221882) [inline Week3/Assignment5/src/Components/SearchBar/index.js:12]: Can make this an utility logic and keep in utils folder

File: Week3/Assignment5/src/Components/SearchBar/index.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <div className="search__container">
+      <form onSubmit={handleSubmit}>
+        <input
+          className="search__input"
+          placeholder="Search User"
+          value={searchTerm}
+          onChange={(e) => setSearchTerm(e.target.value)}
+        />
+        <button className="search__button">Search</button>
```
Comment (ap221882) [inline Week3/Assignment5/src/Components/SearchBar/index.js:26]: can give type="submit" to this button

**Review by ap221882 — COMMENTED** [review]
Reviewed. Well written. Can add commenting where it is needed.

**Pull request merged**